src/extensions/registries/page-registry.ts

```diff
@@ -1,93 +1,95 @@
 // Extensions-api -> Custom page registration
-import type { PageMenuTarget } from "./page-menu-registry";
 import type React from "react";
+import type { UrlParam } from "../../renderer/navigation/url-param";
+
 import path from "path";
 import { action } from "mobx";
-import { compile } from "path-to-regexp";
 import { BaseRegistry } from "./base-registry";
 import { LensExtension, sanitizeExtensionName } from "../lens-extension";
 import logger from "../../main/logger";
-import { rectify } from "../../common/utils";
 
 export interface PageRegistration {
   /**
-   * Page ID or additional route path to indicate uniqueness within current extension registered pages
-   * Might contain special url placeholders, e.g. "/users/:userId?" (? - marks as optional param)
+   * Page-id, part of of extension's page url, must be unique within same extension
    * When not provided, first registered page without "id" would be used for page-menus without target.pageId for same extension
    */
   id?: string;
+  components: PageComponents;
   /**
-   * Strict route matching to provided page-id, read also: https://reactrouter.com/web/api/NavLink/exact-bool
-   * In case when more than one page registered at same extension "pageId" is required to identify different pages,
-   * It might be useful to provide `exact: true` in some cases to avoid overlapping routes.
-   * Without {exact:true} second page never matches since first page-id/route already includes partial route.
-   * @example const pages = [
-   *  {id: "/users", exact: true},
-   *  {id: "/users/:userId?"}
-   * ]
-   * Pro-tip: registering pages in opposite order will make same effect without "exact".
+   * Registered page params.
+   * Used to generate page url when provided in getExtensionPageUrl()-helper.
    */
-  exact?: boolean;
-  components: PageComponents;
-}
-
-export interface RegisteredPage extends PageRegistration {
-  extensionId: string; // required for compiling registered page to url with page-menu-target to compare
-  routePath: string; // full route-path to registered extension page
+  params?: UrlParam[];
 }
 
 export interface PageComponents {
   Page: React.ComponentType<any>;
 }
 
-export function getExtensionPageUrl<P extends object>({ extensionId, pageId = "", params }: PageMenuTarget<P>): string {
-  const extensionBaseUrl = compile(`/extension/:name`)({
-    name: sanitizeExtensionName(extensionId), // compile only with extension-id first and define base path
-  });
-  const extPageRoutePath = path.posix.join(extensionBaseUrl, pageId);
+export interface PageTarget<P = {}> {
+  extensionId?: string;
+  pageId?: string;
+  params?: Record<string, any | any[]> & P; // default target page params
+}
```

Comment:
what is the point of having `any | any[]`?

Also, what is the point of having `& P` where P has no requirements to be an object. 

I think that the following would be better:
```suggestion
export interface PageTarget<P extends Record<string, any> = {}> {
  extensionId?: string;
  pageId?: string;
  params?: P;
}
```

Reply:
Just for info that the value of param could be an array.

Reply:
Yeah but `any` implies that too. If you really want that then it could be `extends Record<string, any | any[]`

Reply:
Alright, not super important :D